src/webgpu/api/validation/capability_checks/features/texture_formats.spec.ts

```diff
@@ -89,6 +88,45 @@ g.test('texture_view_descriptor')
     });
   });
 
+g.test('canvas_configuration')
+  .desc(
+    `
+  Test configuring a canvas with optional texture formats will fail if the required optional
+  feature is not enabled.
+
+  Note: This test should be skipped until GPUCanvasConfiguration allow optional formats.
+  `
+  )
+  .params(u =>
+    u
+      .combine('format', kOptionalTextureFormats)
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('enable_required_feature', [true, false])
+  )
+  .beforeAllSubcases(t => {
+    const { format, enable_required_feature } = t.params;
+
+    const formatInfo = kTextureFormatInfo[format];
+    if (enable_required_feature) {
+      t.selectDeviceOrSkipTestCase(formatInfo.feature);
+    }
+  })
+  .fn(async t => {
+    const { format, canvasType, enable_required_feature } = t.params;
+
+    const canvas = createCanvas(t, canvasType, 2, 2);
+    const ctx = canvas.getContext('webgpu' as const);
+    assert(ctx !== null, 'Failed to get WebGPU context from canvas');
+
+    t.shouldThrow(enable_required_feature ? false : 'TypeError', () => {
+      ctx.configure({
```

Comment:
If `enable_required_feature` is false, this should throw a TypeError.
But if `enabled_require_feature` is true, this should generate a WebGPU validation error (use `t.expectValidationError`).

I don't know whether Chromium correctly implements this today or not, but it might!

Reply:
ok, let me update this PR like that. Thanks!

Reply:
@kainino0x I've updated this PR according to the comments. PTAL.